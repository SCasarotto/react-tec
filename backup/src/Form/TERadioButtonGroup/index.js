import TERadioButtonGroup from './TERadioButtonGroup'

export default TERadioButtonGroup
