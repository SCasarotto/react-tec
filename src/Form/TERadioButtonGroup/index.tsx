import React from 'react'

import { TELabel, TELabelCustomProps } from '../TELabel'

import {
	Container,
	InputWrapper,
	RowWrapper,
	RadioButton,
	Label,
} from './styledComponents'
import { TERowCustomProps } from '../TERow'
import { TERadioButtonInputProps } from '../TERadioButtonInput'
import { manipulateRowData } from './../../helpers'

interface TERadioButtonData {
	label: string
	value?: string
	key?: string
}
interface TERadioButtonGroupProps
	extends TERowCustomProps,
		TELabelCustomProps,
		TERadioButtonInputProps {
	labelForKey: string
	inputRowSize?: string
	checkedValue: string
	buttonArray: (string | TERadioButtonData)[]
	scrolling?: boolean
}
export const TERadioButtonGroup: React.FC<TERadioButtonGroupProps> = (
	props,
) => {
	const {
		rowSize,
		last,
		title,
		onChange,
		checkedValue,
		buttonArray,
		labelForKey,
		disabled,
		required,
		inputRowSize,
		className = '',
	} = props

	return (
		<Container
			rowSize={rowSize}
			last={last}
			className={`TERadioButtonGroup ${className}`}>
			<TELabel
				disabled={disabled}
				required={required}
				className="TERadioButtonGroupTitle">
				{title}
			</TELabel>
			<InputWrapper className="TERadioButtonGroupInputWrapper">
				{buttonArray &&
					buttonArray.map((rowData) => {
						const { label, value, key } = manipulateRowData({
							rowData,
							labelForKey,
						})

						return (
							<RowWrapper
								key={key}
								rowSize={inputRowSize}
								className="TERadioButtonGroupRowWrapper">
								<RadioButton
									value={value}
									onChange={onChange}
									id={key}
									checked={checkedValue === value}
									disabled={disabled}
									className="TERadioButtonGroupRadioButton"
								/>
								<Label
									htmlFor={key}
									disabled={disabled}
									className="TERadioButtonGroupLabel">
									{label}
								</Label>
							</RowWrapper>
						)
					})}
			</InputWrapper>
		</Container>
	)
}
