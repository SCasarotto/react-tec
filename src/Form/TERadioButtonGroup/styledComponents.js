import styled from 'styled-components'

import TERow from './../TERow'
import TELabel from './../TELabel'
import TERadioButtonInput from './../TERadioButtonInput'

export const Container = styled(TERow)`
	${(props) => {
		return `
            margin-bottom: 10px;
            display: inline-block;
            vertical-align: top;
        `
	}}
`
export const InputWrapper = styled.div`
	${(props) => {
		return `
            display: flex;
            flex-wrap: wrap;
        `
	}}
`

export const RowWrapper = styled(TERow)`
	${(props) => {
		return `
            margin-bottom: 5px;
            display: flex;
            align-items: center;
        `
	}}
`

export const RadioButton = styled(TERadioButtonInput)`
	${(props) => {
		return `
            display: inline-block;
            vertical-align: middle;
            cursor: pointer;
        `
	}}
`
export const Label = styled(TELabel)`
	${(props) => {
		const { theme } = props

		return `
            width: auto;
            display: inline-block;
            vertical-align: middle;
            padding-left: 10px;
            margin-bottom: 0px;
            cursor: pointer;
            font-size: 14px;

            @media (max-width: 550px) {
                font-size: 12px;
            }
        `
	}}
`
