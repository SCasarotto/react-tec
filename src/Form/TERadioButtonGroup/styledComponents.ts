import styled from 'styled-components'

import { TERow } from '../TERow'
import { TELabel } from '../TELabel'
import { TERadioButtonInput } from '../TERadioButtonInput'

export const Container = styled(TERow)`
	margin-bottom: 10px;
	display: inline-block;
	vertical-align: top;
`
export const InputWrapper = styled.div`
	display: flex;
	flex-wrap: wrap;
`

export const RowWrapper = styled(TERow)`
	margin-bottom: 5px;
	display: flex;
	align-items: center;
`

export const RadioButton = styled(TERadioButtonInput)`
	display: inline-block;
	vertical-align: middle;
	cursor: pointer;
`
export const Label = styled(TELabel)`
	width: auto;
	display: inline-block;
	vertical-align: middle;
	padding-left: 10px;
	margin-bottom: 0px;
	cursor: pointer;
	font-size: 14px;

	@media (max-width: 550px) {
		font-size: 12px;
	}
`
