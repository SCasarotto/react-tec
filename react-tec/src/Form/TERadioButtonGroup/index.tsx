import React from 'react'

import TELabel, { TELabelCustomProps } from '../TELabel'

import {
	Container,
	InputWrapper,
	RowWrapper,
	RadioButton,
	Label,
} from './styledComponents'
import { TERowCustomProps } from 'Form/TERow'
import { TERadioButtonInputProps } from 'Form/TERadioButtonInput'

interface TERadioButtonData {
	label: string
	value?: string
	key?: string
}
interface ManipulateRowData {
	rowData: string | TERadioButtonData
	labelForKey: string
}
const manipulateRowData = ({
	rowData,
	labelForKey = '',
}: ManipulateRowData) => {
	//Allowing For Greater Shorthand
	if (typeof rowData === 'string') {
		const label = rowData
		const value = rowData
		const key = labelForKey + rowData
		return { label, value, key }
	} else if (typeof rowData === 'object') {
		let { label, value, key } = rowData

		if (!value) {
			value = label
		}
		if (!key) {
			key = labelForKey + label
		}

		return { label, value, key }
	}
	return rowData
}
interface TERadioButtonGroupProps
	extends TERowCustomProps,
		TELabelCustomProps,
		TERadioButtonInputProps {
	labelForKey: string
	inputRowSize?: string
	checkedValue: string
	buttonArray: (string | TERadioButtonData)[]
	scrolling?: boolean
}
const TERadioButtonGroup: React.FC<TERadioButtonGroupProps> = (props) => {
	const {
		rowSize,
		last,
		title,
		onChange,
		checkedValue,
		buttonArray,
		labelForKey,
		disabled,
		required,
		inputRowSize,
		className = '',
	} = props

	return (
		<Container
			rowSize={rowSize}
			last={last}
			className={`TERadioButtonGroup ${className}`}>
			<TELabel
				disabled={disabled}
				required={required}
				className="TERadioButtonGroupTitle">
				{title}
			</TELabel>
			<InputWrapper className="TERadioButtonGroupInputWrapper">
				{buttonArray &&
					buttonArray.map((rowData) => {
						const { label, value, key } = manipulateRowData({
							rowData,
							labelForKey,
						})

						return (
							<RowWrapper
								key={key}
								rowSize={inputRowSize}
								className="TERadioButtonGroupRowWrapper">
								<RadioButton
									value={value}
									onChange={onChange}
									id={key}
									checked={checkedValue === value}
									disabled={disabled}
									className="TERadioButtonGroupRadioButton"
								/>
								<Label
									htmlFor={key}
									disabled={disabled}
									className="TERadioButtonGroupLabel">
									{label}
								</Label>
							</RowWrapper>
						)
					})}
			</InputWrapper>
		</Container>
	)
}

export default TERadioButtonGroup
