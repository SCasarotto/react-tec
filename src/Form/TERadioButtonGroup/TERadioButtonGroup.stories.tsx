import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TERadioButtonGroup, TERadioButtonGroupProps } from './TERadioButtonGroup'

export default {
	title: 'Form/TERadioButtonGroup',
	component: TERadioButtonGroup,
	argTypes: {},
} as Meta

const Template: Story<TERadioButtonGroupProps> = (args) => <TERadioButtonGroup {...args} />

export const Default = Template.bind({})
Default.args = {
	labelForKey: 'TERadioButtonGroup',
	title: 'My Radio Button Group',
	onChange: (e) => console.log(e.target.value),
	buttonArray: ['value1', 'valueTwo', 'OtherValue'],
}
