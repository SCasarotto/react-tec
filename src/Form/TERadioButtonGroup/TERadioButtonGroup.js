//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import TELabel from './../TELabel'

import { Container, InputWrapper, RowWrarpper, RadioButton, Label } from './styledComponents'

const manipulateRowData = ({ rowData, labelForKey = '' }) => {
    //Allowing For Greater Shorthand
    if (typeof rowData === 'string') {
        const label = rowData
        const value = rowData
        const key = labelForKey + rowData
        return { label, value, key }
    } else if (typeof rowData === 'object') {
        let { label, value: value = label, key: key = labelForKey + label } = rowData
        return { label, value, key }
    }
    return rowData
}
const TERadioButtonGroup = (props) => {
    const {
        size,
        last,
        title,
        onChange,
        checkedValue,
        buttonArray,
        labelForKey,
        disabled,
        required,
        rowSize,
        className,
    } = props

    return (
        <Container size={size} last={last} className={className}>
            <TELabel disabled={disabled} required={required}>
                {title}
            </TELabel>
            <InputWrapper>
                {buttonArray &&
                    buttonArray.map((rowData) => {
                        const { label, value, key } = manipulateRowData({ rowData, labelForKey })

                        return (
                            <RowWrarpper key={key} size={rowSize}>
                                <RadioButton
                                    value={value}
                                    onChange={onChange}
                                    id={key}
                                    checked={checkedValue === value}
                                    disabled={disabled}
                                />
                                <Label htmlFor={key} disabled={disabled}>
                                    {label}
                                </Label>
                            </RowWrarpper>
                        )
                    })}
            </InputWrapper>
        </Container>
    )
}

TERadioButtonGroup.propTypes = {
    title: PropTypes.string,
    onChange: PropTypes.func,
    labelForKey: PropTypes.string,
}

export default TERadioButtonGroup
