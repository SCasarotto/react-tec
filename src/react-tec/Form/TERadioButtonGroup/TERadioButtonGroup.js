//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import TELabel from './../TELabel'

import { Container, InputWrapper, RowWrarpper, RadioButton, Label } from './styledComponents'

const TERadioButtonGroup = (props) => {
    const manipulateRowData = ({ rowData, labelForKey = '' }) => {
        //Allowing For Greater Shorthand
        if (typeof rowData === 'string') {
            const label = rowData
            const value = rowData
            const key = labelForKey + rowData
            return { label, value, key }
        } else if (typeof rowData === 'object') {
            let { label, value: value = label, key: key = labelForKey + label } = rowData
            return { label, value, key }
        }
        return rowData
    }

    const {
        size,
        last,
        title,
        onChange,
        checkedValue,
        buttonArray,
        labelForKey,
        disabled,
        required,
        rowSize,
    } = props

    return (
        <Container size={size} last={last}>
            <TELabel disabled={disabled} required={required}>
                {title}
            </TELabel>
            <InputWrapper>
                {buttonArray &&
                    buttonArray.map((rowData) => {
                        const { label, value, key } = manipulateRowData({ rowData, labelForKey })

                        return (
                            <RowWrarpper key={key} size={rowSize}>
                                <RadioButton
                                    value={value}
                                    onChange={onChange}
                                    id={key}
                                    checked={checkedValue === value}
                                    disabled={disabled}
                                />
                                <Label htmlFor={key} disabled={disabled}>
                                    {label}
                                </Label>
                            </RowWrarpper>
                        )
                    })}
            </InputWrapper>
        </Container>
    )
}

TERadioButtonGroup.propTypes = {
    title: PropTypes.string,
    onChange: PropTypes.func,
    labelForKey: PropTypes.string,
}

export default TERadioButtonGroup
