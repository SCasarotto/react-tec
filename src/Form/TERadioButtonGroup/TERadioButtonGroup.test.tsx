import React from 'react'
import { render } from '@testing-library/react'
import { TERadioButtonGroup } from './TERadioButtonGroup'

describe('TERadioButtonGroup', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(
			<TERadioButtonGroup
				labelForKey='testTERadioButtonGroup'
				checkedValue='someValue'
				buttonArray={['someValue', 'otherValue']}
				onChange={() => {}}
			/>,
		)
		expect(asFragment()).toMatchSnapshot()
	})
})
