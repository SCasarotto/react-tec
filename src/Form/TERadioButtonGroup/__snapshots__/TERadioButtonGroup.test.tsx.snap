// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TERadioButtonGroup matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gtssRu cWtTpP TERow sc-hKFyIo lllBvc TERadioButtonGroup "
  >
    <div
      class="sc-eCApGN dpekhW TERadioButtonGroupInputWrapper"
    >
      <div
        class="sc-gtssRu cWtTpP TERow sc-jSFkmK dWyQXW TERadioButtonGroupRowWrapper"
      >
        <input
          checked=""
          class="sc-dlnjPT lmUFUT TERadioButtonInput sc-gKAblj IsMNB TERadioButtonGroupRadioButton"
          id="testTERadioButtonGroupsomeValue"
          type="radio"
          value="someValue"
        />
        <label
          class="sc-bdnylx cRNJdj TELabel sc-iCoHVE ktbxmj TERadioButtonGroupLabel"
          for="testTERadioButtonGroupsomeValue"
        >
          someValue
        </label>
      </div>
      <div
        class="sc-gtssRu cWtTpP TERow sc-jSFkmK dWyQXW TERadioButtonGroupRowWrapper"
      >
        <input
          class="sc-dlnjPT lmUFUT TERadioButtonInput sc-gKAblj IsMNB TERadioButtonGroupRadioButton"
          id="testTERadioButtonGroupotherValue"
          type="radio"
          value="otherValue"
        />
        <label
          class="sc-bdnylx cRNJdj TELabel sc-iCoHVE ktbxmj TERadioButtonGroupLabel"
          for="testTERadioButtonGroupotherValue"
        >
          otherValue
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
